frontend/src/components/UpgradesContainer.tsx

```diff
@@ -31,15 +37,22 @@ export default function UpgradesContainer() {
 
   return (
     <div className="upgrades-container" style={{ marginBottom: "20px" }}>
-      {upgrades.map((up) => (
-        <Upgrade
-          up={up}
-          status={getStatusForUpgrade(up)}
-          key={up.name}
-          onBuy={attemptPurchase}
-          currencyAmount={gameData.currencyAmount}
-        ></Upgrade>
-      ))}
+      {upgrades
+        .filter((up) => {
+          if (getStatusForUpgrade(up) === UpgradeStatus.Hidden) {
```

Comment:
This lambda change be changed to the one liner `.filter((up) => getStatusForUpgrade(up) !== UpgradeStatus.Hidden)`